### DISCUSSION TITLE
getActivePinia was called with no active Pinia. Did you forget to install pinia?

### DISCUSSION THREAD

**Discussion by bjorntj** [vuejs/pinia#971] [Help and Questions] [answered]
Hi.

I am trying to convert to typescript but I get this problem when access the page:

```
getActivePinia was called with no active Pinia. Did you forget to install pinia?
	const pinia = createPinia()
	app.use(pinia)
This will fail in production.
```

I have the following in my main.ts file:```

```
import { createApp } from 'vue'
import App from './App.vue'
import router from './router'
import './assets/tailwind.css'
import { createPinia } from 'pinia'
import piniaPluginPersistedstate from 'pinia-plugin-persistedstate'
import { useAuthStore } from "@/store/auth";
import { useCommonStore } from "@/store/common";
/*import setupInterceptors from '@/services/setupInterceptors'*/

const pinia = createPinia();
pinia.use(piniaPluginPersistedstate);

const app = createApp(App)
/*app.config.productionTip = false*/
app.use(pinia)
app.use(router)

const authStore = useAuthStore();
const commonStore = useCommonStore();

/*setupInterceptors(authStore, commonStore);*/

app.mount('#app')
```

If I comment out the 2 lines that use the store, then I do not get any errors.
What am I missing? I am using the stores after app.use(pinia);

**Comment by scambier** [comment, marked as answer]
@bjorntj  did you solve this error?

Ok, it's most likely because you're calling `useAuthStore()` from outside a component's setup, and it's called before Pinia is setup. https://github.com/vuejs/pinia/discussions/664#discussioncomment-1329898

**Reply by bjorntj** [reply]
Yes, I figured it out... :)

**Reply by ShuvobrotoBiswas** [reply]
then?

**Reply by AndrianD** [reply]
then @bjorntj wants you to rack your brains like him to find the solution, that's the help he's suggesting I think

**Comment by xiaoliuxuesheng** [comment]
I have the same problem
<img width="726" alt="image" src="https://user-images.githubusercontent.com/33746658/153706826-292c0780-5241-49d5-955e-0af554b83aa8.png">

Vite + Vue project: This error occurs when using `store` in `router.beforeeach` after installing `router` and `pinia`
```ts
import { createRouter, createWebHistory, RouteRecordRaw } from 'vue-router'
import { UserStore } from '@/store/UserStore'

const userStore = UserStore()
const routes: RouteRecordRaw[] = []
const router = createRouter({
  history: createWebHistory(),
  routes
})
router.beforeEach((to, from, next) => {
  console.log(userStore)
  next()
})
export default router
```

**Reply by scambier** [reply]
That's because you call `UserStore()` at the root level. If you add some `console.log()`s, you'll lilely see that the call to `UserStore()` occurs before `app.use(pinia)`

**Reply by xufromxu** [reply]
> 

so how to use pinia in router file?
 got it on Pinia official website
https://pinia.vuejs.org/core-concepts/outside-component-usage.html#single-page-applications

**Reply by xiaoliuxuesheng** [reply]
Yes, that's how I solved it, and THEN I looked it up in the document

**Reply by zhoufanglu** [reply]
> > 
> 
> so how to use pinia in router file? got it on Pinia official website https://pinia.vuejs.org/core-concepts/outside-component-usage.html#single-page-applications

thank you share this document, i solve this problem
* put `store` in `beforeEach`
```js
router.beforeEach((to) => {
  // ✅ This will work because the router starts its navigation after
  // the router is installed and pinia will be installed too
  const store = useStore()

  if (to.meta.requiresAuth && !store.isLoggedIn) return '/login'
})
```

**Reply by maicWorkGithub** [reply]
How to use store in web worker? Inside `onmessage` `const store = useStore()` still get same error.

**Reply by arm5472** [reply]
How to use store to **_create_** routes? I need to use store within **createRouter** to create routes array and I get the same error too.

**Reply by ilovepixelart** [reply]
If you still using Vue2 with Vite
Make sure you use PiniaVuePlugin before you use router

router.js or routes/index.js depending on your setup
```TypeScript
import { PiniaVuePlugin } from 'pinia';

Vue.use(PiniaVuePlugin);
Vue.use(Router);
```

**Comment by zulfio** [comment]
In my case, I'm using the script setup style and forgot to write **setup** inside the opening script tag.

from this:
**<script>**
import { useUserDataStore } from "../../stores/UserData";

const userDataStore = useUserDataStore();
</script>

should be like this:
**<script setup>**
import { useUserDataStore } from "../../stores/UserData";

const userDataStore = useUserDataStore();
</script>

**Reply by blackspike** [reply]
I also did this lol

**Reply by IamJim92** [reply]
This is a small detail, but a massively helpful one! - really helped me, thankyou

**Reply by luisaromero** [reply]
thanks !! it helps me

**Reply by Sharmacs8853** [reply]
thanks !! it helps

**Reply by mariana-morais** [reply]
thanks

**Reply by SantiPorres** [reply]
Thanks a lot!!!

**Reply by wawanneutron** [reply]
thanks brother

**Reply by PeterPilley** [reply]
This was the answer!!!. Been trying to solve this for a couple of days now thanks!!!!

**Reply by ccf112** [reply]
thanks

**Comment by lead-fullstack-developer** [comment]
Hi, everyone after a lot of research I found the answer to this issue,
you must pass index.ts/js for const like below:


`<script lang="ts" setup>
import store from '../stores/index';
import { useCounterStore } from '../stores/counter';

const counterStore = useCounterStore(store());
counterStore.increment();
console.log(counterStore.count);
</script>
`

**Reply by barbirict** [reply]
Thanks, you saved me three hours of pain haha. I'd like to add that all it took to fix it was changing
```javascript
<script lang="js" setup>
```
to

```javascript
<script lang="ts" setup>
```
Not even sure why I had it set to js lol.

**Reply by lead-fullstack-developer** [reply]
nice

**Reply by ccosse** [reply]
fwiw this never happened for me until I upgraded quasar-cli from version 1 to version 2.
Initializing with store() as arg to useStore() is only way it works for me now.

**Comment by 1714080902120** [comment]
i had the same problem when i used `vitest` and test a component outside `src`...
so i have to import main.ts in all test file before run test

**Reply by dxphilo** [reply]
Having the same issue here. I am testing a component that uses the pinia store. Unfortunately not sure what might be off. Testing it like so but its throwing the error.

```
import { describe, expect, test } from 'vitest'
import { mount } from '@vue/test-utils'
import Prompt from '../components/Prompt.vue'

describe('Promp component is mounted', () => {
  test('is mounted', () => {
    const wrapper = mount(Prompt)
    expect(wrapper.exists()).toBe(true)
  })
})
```

**Reply by amycai-arcadis** [reply]
I am having the same problem here, did you figure it out?

**Reply by amycai-arcadis** [reply]
I fixed it with:
import { setActivePinia, createPinia } from 'pinia'

and add:
  beforeEach(() => {
    // creates a fresh pinia and make it active so it's automatically picked
    // up by any useStore() call without having to pass it to it:
    // `useStore(pinia)`
    setActivePinia(createPinia())
  })

Full test case:
// stores/counter.spec.ts
import { setActivePinia, createPinia } from 'pinia'
import { useCounter } from '../src/stores/counter'

describe('Counter Store', () => {
  beforeEach(() => {
    // creates a fresh pinia and make it active so it's automatically picked
    // up by any useStore() call without having to pass it to it:
    // `useStore(pinia)`
    setActivePinia(createPinia())
  })

  it('increments', () => {
    const counter = useCounter()
    expect(counter.n).toBe(0)
    counter.increment()
    expect(counter.n).toBe(1)
  })

  it('increments by amount', () => {
    const counter = useCounter()
    counter.increment(10)
    expect(counter.n).toBe(10)
  })
})

check the documentation here:
https://pinia.vuejs.org/cookbook/testing.html

**Reply by dxphilo** [reply]
Thanks

**Reply by franciscobaca** [reply]
You are a god. Thank you!

**Reply by MichaelAffanyi** [reply]
setActivePinia and createPinia didn't work for me. Anyone has another solution?

**Reply by anibalsanchez** [reply]
I think that the source of the problem is a problem with the dependencies. Something breaks in Pinia bundling, and the internal variables aren't correctly set.

**Comment by LasanthaKathriarachchi** [comment]
Hi Guys,

I created a tableau extension, in which I want to render a map component using pinia.
Coding done using Vue + TS + JS.

This is how I try to load the app with map in the main.js file.

------------
main.js
------------

import { createApp } from 'vue';
import { createPinia } from 'pinia';
import App from './App.vue';
import "./styles/general.scss";
import directives from "./directives";

const app = createApp(App);
directives(app);
const pinia = createPinia();
app.use(pinia);
app.mount('#app');

--------------------------------------------

I'm getting this error.

-------------------------------
getActivePinia was called with no active Pinia. Did you forget to install pinia?
const pinia = createPinia()
app.use(pinia)
This will fail in production.

-----------------------------

Please let me know, any thing more require to render the pinia map inside a tableau extension.

Thanks in advance,
Lasantha

**Comment by nukuutos** [comment]
Hi, guys maybe it will be helpful. 

I encountered this problem in `typescript` project.   
I created `pinia` store with `.js` extension: `my-store.js` and got this:
```getActivePinia was called with no active Pinia. Did you forget to install pinia?```

After changing filename extension to `my-store.ts` everything works well

**Comment by LasanthaKathriarachchi** [comment]
Hi Nikita Voloshin.

You are correct.
When the map-store as a ts file, it works. I tried it that way in a powerbi custom visual. There it didn't gave this error.
But when the same code used in a Tableau Extension, it needed to compile all the ts files into js files. 
Then it doesn't work and gave this error.

Anybody knows how to solve the error, when the map-store used as a js file?

**Comment by clifford-devops3** [comment]
I had the same issue and the reason was that I had not included the **setup** keyword in my script

**Reply by DreiBelle** [reply]
how about mine/
// Navigation guard
router.beforeEach((to, from, next) => {
  const authStore = useAuthStore(); // Replace this with your actual store or authentication logic

  // Check if the route requires authentication
  if (to.meta.requiresAuth) {
    // Check if the user is authenticated
    if (authStore.isAuthenticated) {
      // User is authenticated, allow navigation
      next();
    } else {
      // User is not authenticated, redirect to the login page
      next('/Login'); // Change '/Login' to your actual login route
    }
  } else {
    // If the route does not require authentication, allow navigation
    next();
  }
});

**Comment by jadsow** [comment]
I have the same problem.
I use setup on my components.

My index store:
![image](https://github.com/vuejs/pinia/assets/95438480/c3d54524-0ecd-4203-9a4e-982d667ce2c6)


My store:
![image](https://github.com/vuejs/pinia/assets/95438480/4ad68906-8a29-4692-8f5f-a944833fe6a4)

My component:
![image](https://github.com/vuejs/pinia/assets/95438480/9983f82f-9cb4-43e3-bcb8-f529639ad618)

Error:
![image](https://github.com/vuejs/pinia/assets/95438480/4234db82-27b3-4439-8d4b-3e9c0f793ec4)

Can anybody help me please?

**Reply by svenhue** [reply]
create a quasar bootfile like this:

```
import { boot } from 'quasar/wrappers';
import { createPinia } from 'pinia';

// Be careful when using SSR for cross-request state pollution
// due to creating a Singleton instance here;
// If any client changes this (global) instance, it might be a
// good idea to move this instance creation inside of the
// "export default () => {}" function below (which runs individually
// for each client)

export default boot(({ app }) => {
  const pinia = createPinia();
  pinia.install(app);
});
```

And dont forget to register this file in your quasar.config.

**Comment by Catcheer** [comment]
in my case in the router file use
```
 component:  () => import('../views/UserInfo.vue')
```
instaed

```
import User from '../views/UserInfo.vue'
```

solved my promble

**Reply by perfumescent** [reply]
thank you!!!!!!!!!!!!!!!!

**Comment by mirrors-cl** [comment]
import { createPinia } from 'pinia';
import { useCounterStore } from '@/stores/counter';


const pinia = createPinia();
const store = useCounterStore(pinia);
console.log('storedata',store.orderData)

**Comment by anibalsanchez** [comment]
Today, I had to add this (in a project that has been working fine for years):

```ts
  useAppStore(pinia)
  app.mount('#app')
```

**Comment by cipi1965** [comment]
For me it was a mixed use of js and ts files
